compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/indy/LambdaMetafactoryArguments.kt

```diff
@@ -82,6 +83,10 @@ internal class LambdaMetafactoryArgumentsBuilder(
         if (context.irIntrinsics.getIntrinsic(implFun.symbol) != null)
             return null
 
+        // Don't generate reference to inlineOnly methods as those do not exist at runtime.
+        if (implFun.hasAnnotation(INLINE_ONLY_ANNOTATION_FQ_NAME))
```

Comment:
This should probably be at least `implFun.isEffectivelyInlineOnly()`, because invokedynamic won't work if the function has reified type parameters.

```
import java.util.function.Consumer

inline fun <reified T> foo(x: T) =
    println("${T::class.simpleName}($x)")

fun call1(c: Consumer<String>, s: String) {
    c.accept(s)
}

fun main() {
    call1(::foo, "Hello World!") // java.lang.UnsupportedOperationException: This function has a reified type parameter and thus can only be inlined at compilation time, not called directly.
}
```

Plus, if the function is a property accessor, the property has the annotation, not the function (`isEffectivelyInlineOnly` handles that too).

Maybe it's best not to use invokedynamic with inline functions at all so that they are indeed copied as before?